### ISSUE TITLE
Document how to build the EF fort

### ISSUE THREAD

**Issue by sebastienros** [aspnet/EntityFrameworkCore#11142]
Couldn't find any documentation on how to rebuild it when it's broken. I found some filled cans, was not sure where to put them. Also some clear schema with the don't matrix for the E and the F would be useful.

Are there some tests I can run to ensure its done correctly?

![Sebastien without docs](https://user-images.githubusercontent.com/1430078/36998198-f89c1f9e-2070-11e8-9b70-f67144bbdfd2.jpg)

**Comment by divega** [comment]
@sebastienros I thought you had already tested it, and it didn’t go well. Next time just some light smoke testing please :smile:

**Comment by DamianEdwards** [comment]
Any docs need to take optimal performance into consideration too #perfculture

**Comment by Eilon** [comment]
Here are some suggestions on other patterns-based designs that could be used:

The original, ~6.5' (2 meters) Tower of Coke:

![image](https://user-images.githubusercontent.com/202643/36951786-489c8dc2-1fbc-11e8-9316-2c30c5a8c51a.png)

The revised, ~9' (2.7 meters) Tower of Coke:

![image](https://user-images.githubusercontent.com/202643/36951787-4dfc90d2-1fbc-11e8-9032-459f246700fd.png)

**Comment by bricelam** [comment]
Cherry Coke hasn't tasted the same since they changed from this epic can design...

**Comment by Eilon** [comment]
@bricelam so true... so true...

**Comment by roji** [comment]
!!!

**Comment by Eilon** [comment]
@roji I believe I told you I'd try to find the pictures... those are the pictures 😄

**Comment by roji** [comment]
And I thought we were doing so well...

**Comment by Eilon** [comment]
@roji oh you were doing great, just with a different set of objectives 😄

**Comment by roji** [comment]
Those are kind words @Eilon but I'm just gonna be depressed in the corner now, thanks.

**Comment by ajcvickers** [comment]
Blocked on lack of glue.

**Comment by Eilon** [comment]
I found that regular super glue works, but is not very effective. It is too "thin," and doesn't adhere well to the shiny surface of the cans.

Some sort of thicker cement would be ideal. We'd need to run some structural tests on glued sections to figure out what works best.

**Comment by roji** [comment]
Label area-tools? Definitely avoid up-for-grabs, or at least add a providers-beware in case any are standing around...

**Comment by ajcvickers** [comment]
Before (only picture we have is after some damage):
![before1](https://user-images.githubusercontent.com/1430078/36998199-f8b0e2a8-2070-11e8-9a5d-f2c6e7d77b84.jpg)
Crash dump:
![crash](https://user-images.githubusercontent.com/1430078/36998198-f89c1f9e-2070-11e8-9b70-f67144bbdfd2.jpg)
After re-build:
![after2](https://user-images.githubusercontent.com/1430078/36998200-f8c57c7c-2070-11e8-9e2f-d577b45f40d2.jpg)
![after1](https://user-images.githubusercontent.com/1430078/36998201-f8d92be6-2070-11e8-8346-76837d6e05f5.jpg)

**Comment by ajcvickers** [comment]
Initial observations from analysis of re-build:
* It's not in the middle of the room
* It's not straight
* There facing discrepancies for some of the cans.

**Comment by pmiddleton** [comment]
The EF team definitely needs a tack welder in the office.

**Comment by Eilon** [comment]
Tack welding aluminum is not trivial, apparently. But if we put our minds to it... we can definitely start a huge fire! 😄

**Comment by roji** [comment]
You broke it and built it again from scratch? You guys need to add "core" to it then... Only a few more cans...!

**Comment by ajcvickers** [comment]
A first attempt to add structural integrity (Scotch tape) resulted in disaster. We really need to hire a hardware engineer. Software engineers building real-world stuff isn't working too well...

![ohno](https://user-images.githubusercontent.com/1430078/37055918-c50b63d4-2137-11e8-90d3-2fd3fae4bfdb.jpg)

**Comment by BojidarStanchev** [comment]
Just today I thought "What are Microsoft doing that is so important that they ignore EF core Many2Many implementation for so long". Then I saw this.

**Comment by pmiddleton** [comment]
That is EF Core Many 2 Many - Coke, Pepsi, Dr Pepper, Mountain Dew....

**Comment by ajcvickers** [comment]
Current status: reconstructed, realigned, but still no glue:
![noglue2](https://user-images.githubusercontent.com/1430078/37179176-a10033c2-22d9-11e8-98f7-7cc9925ff994.jpg)
![noglue1](https://user-images.githubusercontent.com/1430078/37179126-7561ed50-22d9-11e8-81df-4be3bda82eb1.jpg)

**Comment by blowdart** [comment]
I find it incredulous that no blame was laid for this destruction when, as ever, the fault lies with @davidfowl (Junior) who apparently, when I was away, removed the security from my desk, borrowed some toys and flung them with no apparent planning or threat model. He takes after his dad that one.

**Comment by ajcvickers** [comment]
@blowdart Well...I heard that he was encouraged by a certain French gentleman. At least some of the blame must lie there. After all, if we're not responsible adults in our own workplace, then how can we expect the younger generation to learn appropriate behavior?

**Comment by blowdart** [comment]
From a security POV I will point out that I firmly instructed the young man in the proper use of my pokeballs, which was to throw them at his daddy _ONLY_. Unapproved usage cannot be secured.

**Comment by pmiddleton** [comment]
Has JB Weld been considered?

**Comment by blowdart** [comment]
Also from a quick examination the initial vulnerability still exists and  the current implementation remains insecure and unprotected.

It may be time for a comprehensive Sodacan Defense Lifecycle (SDL) review

![Insecure EF](https://i.imgur.com/UAJoHGi.jpg "Insecure EF")

**Comment by ajcvickers** [comment]
@AndriySvyryd to post final picture.

**Comment by pmiddleton** [comment]
Will there be Lego style instructions so we can build our own at home?

**Comment by AndriySvyryd** [comment]
@pmiddleton You can see the source on some of the photos and there should be no licensing issues if you just copy that.

**Comment by AndriySvyryd** [comment]
![wp_20180605_15_40_10_pro](https://user-images.githubusercontent.com/6539701/41006970-86c00970-68d8-11e8-892f-394e7d2dcec5.jpg)

**Comment by Eilon** [comment]
Let's not rule this out... http://mynorthwest.com/category/earthquake_tracker/

**Comment by aspnet-hello** [comment]
We periodically close 'discussion' issues that have not been updated in a long period of time.

We apologize if this causes any inconvenience. We ask that if you are still encountering an issue, please log a new issue with updated information and we will investigate.
